react-native/react/constants/platform.native.desktop.js

```diff
@@ -20,26 +20,43 @@ const envedPathOSX = {
   prod: 'Keybase'
 }
 
+function buildWin32SocketRoot() {
+  let appdata = process.env.APPDATA
+  // Remove leading drive letter e.g. C:
+  if (/^[a-zA-Z]:/.test(appdata)) {
+    appdata = appdata.slice(2)
+  }
+  // Handle runModes, prod has no extension.
+  let extension = ''
+  if (runMode !== 'production') {
+    extension = runMode.charAt(0).toUpperCase() + runMode.substr(1)
+  }
+  let path = `\\\\.\\pipe\\kbservice${appdata}\\Keybase${extension}`
+  return path
+}
+
 function findSocketRoot () {
   const paths = {
     'darwin': `${process.env.HOME}/Library/Caches/${envedPathOSX[runMode]}/`,
-    'linux': runMode === 'prod' ? `${process.env.XDG_RUNTIME_DIR}/keybase/` : `${process.env.XDG_RUNTIME_DIR}/keybase.${runMode}/`
+    'linux': runMode === 'prod' ? `${process.env.XDG_RUNTIME_DIR}/keybase/` : `${process.env.XDG_RUNTIME_DIR}/keybase.${runMode}/`,
```

Comment:
Take a look at the docs. You pass a param

On Tuesday, December 22, 2015, Chris Ball <notifications@github.com> wrote:

> In react-native/react/constants/platform.native.desktop.js
> <https://github.com/keybase/client/pull/1591#discussion_r48294686>:
>
> >  function findSocketRoot () {
> >    const paths = {
> >      'darwin': `${process.env.HOME}/Library/Caches/${envedPathOSX[runMode]}/`,
> > -    'linux': runMode === 'prod' ? `${process.env.XDG_RUNTIME_DIR}/keybase/` : `${process.env.XDG_RUNTIME_DIR}/keybase.${runMode}/`
> > +    'linux': runMode === 'prod' ? `${process.env.XDG_RUNTIME_DIR}/keybase/` : `${process.env.XDG_RUNTIME_DIR}/keybase.${runMode}/`,
>
> No -- getAppPath() is the directory where Electron is installed, which is
> totally different to the place we can find the service's socket. To find
> the service's socket, we need to duplicate its algorithm for placing the
> socket, and it uses the same XDG/etc stuff.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/client/pull/1591/files#r48294686>.
>